### ISSUE TITLE
CS8601/CS8625 False positive

### ISSUE THREAD

**Issue by Logerfo** [dotnet/roslyn#40728]
## Version Used
Visual Studio 2019 (16.5.0 Preview 1.0)

## Steps to Reproduce
```c#
using System.Diagnostics.CodeAnalysis;
#nullable enable
public class C {
    [AllowNull]
    public string Text { get; set; } = "";
    public void M() {
        var c = new C();
        c.Text = null;
        new C
        {
            Text = null // CS8625
        };
    }
}
```
[SharpLab](https://sharplab.io/#v2:CYLg1APgAgDABFAjAOgCIEsCGBzAdgewGcAXdAY0OQGF9gBTAQV0wBsBPQ9QgbgFgAoAMS4ArixaYARizpw6zaXQFQAzAgBMcKnADeAuAbgBtBuPwB3AHJiWAXX2HVCRPAAqdAB7FdcbHWLccIT+gQC+cAC8cABE0Xz8hghqUAAscACyABQAlLoOiQYAbpgATnBkkXC4dOZaOfEFhmTI7l6VouINjVU1WvkFegndia3eUR0s/YmhXQahAqFAA===)

## Expected Behavior
Since the behavior for assigning after construction and inside the initializer should be the same, no warnings should be presented.

## Actual Behavior
> Warning CS8625: Cannot convert null literal to non-nullable reference type.

**Comment by jcouv** [comment]
Verified that the issue was fixed. It no longer repro's in 16.5p3. I'll go ahead and close. Thanks

![image](https://user-images.githubusercontent.com/12466233/73404153-dfcccf00-42a5-11ea-8db2-3530ff46ea30.png)
